### PR TITLE
Adding 'Nomai Grand Prix' button that loads into random spawn, and maintaining defualt behavior for 'Resume Expedition'

### LINKED ISSUE #1: Make menu item to "turn on the mod"
call it something funny like "Nomai Grand Prix". For this task, the mod shouldn't do anything if you pick resume expedition, should only do mod things if you pick the new menu item.

### PR DESCRIPTION
**Pull request by cmaines**
This change does the following:
- Adds event to listen to events when player returns to title screen & unsets _isGameStarted flag.
- Changes "_onceFlag" to "_shouldWarp", reversing the behavior of the flag so it is unset after a desired warp occurs.
- Re-sets various tagged GameObjects when game is exited & re-entered.
- Adds a "Nomai Grand Prix" button that spawns the player in a random location.
- Reverts to the default behavior of the "Resume Expedition" button.

Testing performed:
- Verified 'Resume Expedition' enters the normal game.
- Verified 'Nomai Grand Prix' warps the player to a random location.
- Verified exiting the game and clicking 'Nomai Grand Prix' again warps the player to another random location.
- Verified existing the game and clicking 'Resume Expedition' after having clicked 'Nomai Grand Prix' still enters the normal game.

NOT tested:
- Behavior of 'Nomai Grand Prix' button when player has no existing save file; this might crash.

### CODE REVIEW

File: OuterWildsRandomSpeedrun/OuterWildsRandomSpeedrun.cs [hunk 1]
```diff
@@ -83,35 +88,43 @@ private void OnEvent(MonoBehaviour behaviour, Events ev)
...
-                ModHelper.Console.WriteLine($"initialize spawner.", MessageType.Info);
-                var spawnPointsField = typeof(PlayerSpawner)
-                    .GetField("_spawnList", BindingFlags.NonPublic | BindingFlags.Instance);
-                var spawnPoints = spawnPointsField?.GetValue(_spawner) as SpawnPoint[];
-                _spawnPoints = spawnPoints.OrderBy(x => x.name).ToArray();
-
-                ModHelper.Console.WriteLine($"Registered {spawnPoints.Length} spawn points", MessageType.Info);
-                _spawnPoint = randomSpawnPoint();
+            InitSpawnPoints();
+            _spawnPoint = getRandomSpawnPoint();
```
Comment (bkirz) [inline OuterWildsRandomSpeedrun/OuterWildsRandomSpeedrun.cs:106]:
```suggestion
            _spawnPoint = GetRandomSpawnPoint();
```

File: OuterWildsRandomSpeedrun/OuterWildsRandomSpeedrun.cs [hunk 2]
```diff
@@ -83,35 +88,43 @@ private void OnEvent(MonoBehaviour behaviour, Events ev)
...
+            ModHelper.Console.WriteLine($"Registered {spawnPoints.Length} spawn points", MessageType.Info);
+        }
+
+        protected void InitSpawner() {
+            ModHelper.Console.WriteLine($"initialize spawner.", MessageType.Info);
+            _spawner = GameObject.FindGameObjectWithTag("Player").GetRequiredComponent<PlayerSpawner>();
         }
 
-        protected SpawnLocation randomSpawnPoint()
+        protected SpawnLocation getRandomSpawnPoint()
```
Comment (bkirz) [inline OuterWildsRandomSpeedrun/OuterWildsRandomSpeedrun.cs:127]:
```suggestion
        protected SpawnLocation GetRandomSpawnPoint()
```

**Review by bkirz — APPROVED** [review]
lgtm! one non-blocking nit on method name casing

**Comment by cmaines** [comment]
Fixed the case changes locally, will include in a follow-up PR.

**Pull request merged**